docs/examples/plot_2D_basis_function.py

```diff
@@ -0,0 +1,154 @@
+# -*- coding: utf-8 -*-
+
+"""
+# Plotting 2D Basis Functions
+
+## Defining a 2D Basis Function
+
+To model the response of a neuron to the position of an animal in an arena, we need a 2D model for stimuli.
+In this case, the stimuli are the 2D coordinates (x, y) representing the animal's position at each time step.
+
+We can model a response function for multi-dimensional stimuli as the external product of two 1D basis functions.
+The response function approximation for the 2D case would be:
+
+$$
+f(x, y) \\approx \sum_{ij} \\alpha_{ij} \, a_i (x) b_j(y).
+$$
+
+Here, we define this 2D basis function as the multiplication of two 1D basis objects.
+"""
+
+import numpy as np
+import matplotlib.pyplot as plt
+import neurostatslib as nsl
+
+# Define 1D basis objects
+a_basis = nsl.basis.MSplineBasis(n_basis_funcs=20, order=3)
+b_basis = nsl.basis.MSplineBasis(n_basis_funcs=10, order=2)
+
+# 2D basis function as the product of the two 1D basis objects
+prod_basis = a_basis * b_basis
+
+# %%
+# Basis Evaluation and Plotting
+# ------------------------------------
+# We evaluate the product of two 1D basis objects which requires two inputs: 
+# one for the x-coordinate and one for the y-coordinate.
+# The number of elements of the product basis is the product of the elements of the 1D basis.
+
+# Define a trajectory with 1000 time-points representing the recorded trajectory of the animal
+T = 1000
+x_coord = np.linspace(0, 10, 1000)
+y_coord = np.linspace(0, 50, 1000)
```

Comment:
Why have them go to different maximum values? Unless there's a specific reason that we explain, make them identical.

Reply:
edited so tat the range is the same